### ISSUE TITLE
Suggestion: Endless pelting by tiny asteroids

### ISSUE THREAD

**Issue by dplepage** [endless-sky/endless-sky#612]
This is a spinoff of a comment I made on #594.

What if some systems had dense clouds of tiny asteroids that would actually hit and damage your ship? 

This would have a number of interesting affects on the gameplay:
 * Systems with high tiny-asteroid counts would be inaccessible to small ships - if you don't have anti-missile systems or sufficient shield regeneration, you'll be torn apart by all the tiny impacts. This provides a gatekeeping mechanism, allowing parts of the galaxy to be blocked off until the player has advanced enough to have a ship capable of surviving the asteroids; jobs to these systems would presumably pay more, and interesting areas furthering various plots could be sealed off this way.
 * Missile combat in tiny-asteroid systems would be fairly different, because anti-missile turrets would be largely occupied with shooting down asteroids (unless, perhaps, a more expensive anti-missile turret were smart enough to focus fire on the most damaging incoming ordnance).
 * Fighters would be worthless in these systems, because they couldn't handle the impacts
 * If the flow were consistent (e.g., the asteroids all orbit the sun instead of flying in random directions), then fleet formations would be really useful - you could actually arrange your ships so that your better-shielded ships take the brunt of the asteroids while your more fragile ships stay in their "asteroid shadow". This could even lead to new ship types, like a short, wide ship with a really thick hull designed specifically to absorb asteroids.
 * If the flow is consistent, then these impacts would also provide a kind of "current" - if you do nothing, your ship will be pushed in circles according to the prevailing flow of asteroids. This would also mean that flying against the flow means you A) move more slowly, and B) take more damage, which means that maneuvering around your opponents while fighting in these systems would be much more strategic.

**Comment by DingusShingleton** [comment]
Asteroid fields in real life are nothing like they are in the movies; even the densest among them would be easily traversable without ever seeing even a single asteroid.

Having said that.... I love this idea! Dense asteroid fields are a staple of space faring games, and all of the mechanics that you mention I think would be really cool. I like the idea of barrier systems, and that got me thinking that a vast, seemingly empty region of space would be a pretty sweet thing. Like an are of 20 or 30 uninhabited systems, and you better have a lot of fuel or plenty of ramscoops. But who knows what you might find in there!?

Off topic, anyway, yes, I agree. I think environmental hazards are a great risk/reward mechanic. Black holes, asteroid fields, red giants that slowly overheat your ship (like in FTL). Cool stuff.

**Comment by thenerdfreak** [comment]
No. I absolutely despise the "gatekeeping" feature. I read to that part and frowned. This game is meant to be open ended, not locked away by "gates"

**Comment by Pointedstick** [comment]
This is a great idea! I envision your point about small ships being implemented, or at least implementable in the opposite fashion: hurting large ships more than small ones; the small ones could be said to present less surface area and be more maneuverable so they would be in proportionately less danger. This would penalize huge ships and provide some much-needed impediments to the natural progression of accumulating more and more larger and larger ships. Could make fighter swarms more useful, too, as I think they're supposed to be.

**Comment by dplepage** [comment]
@DingusShingleton To be fair, the existing asteroid fields are also nothing like real life - they fill an entire solar system, move in completely random directions, and apparently are all about 1/3 the size of the moon yet never strike spaceships and don't exert any gravity, so I'm not too worried about the realism :smile:.

Also, a huge +1 to large empty areas of space. EVO was the first EV I played, and one of my favorite memories of it was finally getting a ship with enough fuel to properly explore the abandoned wastes north of human space, and being astonished to find the entire Miranu civilization out there.

They're a little tricky, though, because you expect very few ships to come through that way, and so if you don't have a ramscoop and get stuck out there, you might be waiting forever for someone who can help you.

**Comment by Pointedstick** [comment]
@dplepage I had the same experience as a kid and remember the experience with great fondness. And besides, this game already has "gates." You can't visit the Korath or the Wanderers (and presumably many others in the future) until you get some jump drives, for example.

**Comment by dplepage** [comment]
@thenerdfreak It wouldn't be locking it away, it'd just stop you from being able to go everywhere right at the beginning. I think this is actually a really important part of exploration-based games - it's always great when there are places you know are reachable, but you don't yet have the tools to reach them, because it gives you something to work towards and a great sense of accomplishment once you can do them. In that sense, it's no different than the "gatekeeping" that stops you from exploring pirate worlds until your ship is tough enough not to die to pirates, or the "gatekeeping" that stops you from exploring Hai space until you have enough fuel to risk exploring that open space.

**Comment by thenerdfreak** [comment]
Korath are not locked by jumpdrives, they're the key to getting them before the storyline. and that isn't really gatekeeping, that's just fuel management and quick landing

**Comment by dplepage** [comment]
@thenerdfreak It's a set of systems you can't get to until you've acquired the right tools to get there, which is exactly the same thing I'm proposing.

**Comment by thenerdfreak** [comment]
As long as by "large enough ship" you mean anything bigger than an interceptor, then that isn't too bad I guess

**Comment by dplepage** [comment]
@thenerdfreak Yeah, I'm thinking it would be nigh-impossible with an interceptor or a light transport (e.g. a Shuttle or a Scout); with a light warship or medium-size transport you'd be able to travel through but you'd have to be quick (because you'd be slowly losing shields the whole time), and with a heavy warship or transport you wouldn't worry much because your shield generators would replenish your shields faster than the asteroids would grind them down.

That said, @Pointedstick makes a good point that it could also go the other way - you could argue that interceptors are small enough to dodge small asteroids, so only large ships take damage from them.

**Comment by thenerdfreak** [comment]
Then the argument could just be made that you're working under the assumption that space is 2-D and that there is no way to maneuver over/under them

**Comment by endless-sky** [comment]
On a practical level, if the "asteroids" can strike any ship in the system then the game has to model a very large number of them, unless they're limited to a very small distance from the system center.

Also, I'm hesitant to add other random ways that you ship can die. Accidentally jumping into pirate crossfire is already annoying enough early in the game. I wonder if there are other ways to block exploration that wouldn't be quite so lethal. For example, maybe there could be hyperspace routes that only become visible if you have a certain type of sensor, or alien hypergates that only open for you if you have the right "key."

**Comment by Pointedstick** [comment]
Honestly, at the current level of wealth progression, the "early game" doesn't last very long. You can get a markedly superior ship much, much faster than you ever could in any of the EV games. "It makes the early game harder" is something that doesn't actually seem like such a bad idea to me.

If this suggestion was implemented in such a way that it predominately affected larger ships, that wouldn't even be a concern, too.

**Comment by dplepage** [comment]
@endless-sky Possibly thick asteroid clouds could be detectable from a distance, so that you could tell before entering an unexplored system that it was likely to be full of them?

**Comment by guillochon** [comment]
+1 for this, except for maybe story reasons I don't think any systems should have a "lethal" amount of asteroids, but it would be really cool to include them as a mild hazard in some systems. You could then sell outfits that shoot down incoming asteroids before they strike your ship. I've always thought of Escape Velocity as having a strong Star Wars influence, and so the asteroid hazards fit right in.

**Comment by Pointedstick** [comment]
Yeah, I like the idea of "mild hazards." It would give the galaxy a lot more flavor and texture.

**Comment by guillochon** [comment]
One other hazard that was in EV was radar scrambling; this would also be something nice to have in the game.

**Comment by bobafetthotmail** [comment]
>"No. I absolutely despise the "gatekeeping" feature. I read to that part and frowned. This game is meant to be open ended, not locked away by "gates""

Gatekeeping is integral to games. Without gates there is no progression. Without gates there is no game. :)

Big ships are gated by $$$ cost (or boarding mechanics), for example.

Hai are gated behind enough jumps that starter ships won't reach them without fuel pods.

Kroath are gated behind jump drives, that are themselves gated behind either "free worlds plot" or "capping a Kroath capital".

Most missions with big pay are also gated by either cargo space, passenger space, or ship power (or ship type, maybe in the future).

Gating must be done, but must be also done with skill, many many many games screw up here, so "gating" gets a bad name.

>"On a practical level, if the "asteroids" can strike any ship in the system then the game has to model a very large number of them, unless they're limited to a very small distance from the system center."

I'd say to just cheat, much easier.

You make a tile of many tiny fake asteroids the same you do with normal asteroids, you remove all their weapon-stopping ability so they are there just for show, and the damage-dealing part is done by a flat area-of-effect system, like "all ships closer than X to center of the effect get Y damage per second".

Might want to add a bunch of random tiny explosion effects on the ships in the asteroid field, to simulate impacts without having "true" impacts. If the tiny asteroids move fast enough and there are enough of them, it won't look weird. Maybe happening at the same time as the ship taking damage.

With this system you can also easily dial the damage to punish harder the capitals while not insta-nuking ligher ships, as you aren't relying on impacts but on a more flexible "damage ships here" mechanic that can also be made to look at what ships are there before calculating damage to apply on each.


Btw, a similar mechanic can be used for other kinds of space weather, solar flares, gravitic anomalies, random magic stuff someone comes up, ah yes, minefields too.

Just swap the effects and change the numbers.

**Comment by endless-sky** [comment]
In the new Korath territory, there's a system that's infected by a nanobot swarm that will try to digest your ship. That's a bit like micro-asteroids, and has the advantage of not making it necessary to track the locations of thousands or tens of thousands of tiny objects.

**Comment by endless-sky** [comment]
Closing this to focus the issues list on feature requests that are more likely to get implemented.
